wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/RegistryContainer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver;
+
+import static com.google.common.collect.Maps.newConcurrentMap;
+
+import com.google.common.collect.ImmutableMap;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.PathMatcher;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.inject.Singleton;
+import org.eclipse.che.api.languageserver.LanguageServerConfig.CommunicationProvider;
+import org.eclipse.che.api.languageserver.LanguageServerConfig.InstanceProvider;
+import org.eclipse.che.commons.lang.Pair;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+/** Instance contains all language server internal registries. */
+@Singleton
+class RegistryContainer {
```

Comment:
Since it has no constructor, is annotated by `@Singleton` and is injected in many other classes, It is initialized automatically by DI container. You can find more [here](https://github.com/google/guice/wiki/GettingStarted).